applications/asset_tracker_v2/src/modules/location_module.c

```diff
@@ -231,48 +231,45 @@ static void search_start(void)
 {
 	int err;
 	struct location_config config;
-	enum location_method methods[3];
 	int methods_count = 0;
-	int methods_index_gnss = -1;
-	int methods_index_cellular = -1;
+	struct location_method_config methods_updated[CONFIG_LOCATION_METHODS_LIST_SIZE] = { 0 };
 
 	if (copy_cfg.no_data.neighbor_cell && copy_cfg.no_data.gnss && copy_cfg.no_data.wifi) {
 		SEND_EVENT(location, LOCATION_MODULE_EVT_DATA_NOT_READY);
 		LOG_ERR("All GNSS, cellular and Wi-Fi are configured off");
 		return;
 	}
 
-	if (!copy_cfg.no_data.gnss) {
-		methods[methods_count] = LOCATION_METHOD_GNSS;
-		methods_index_gnss = methods_count;
-		methods_count++;
-	}
-
-#if defined(CONFIG_LOCATION_METHOD_WIFI)
-	if (!copy_cfg.no_data.wifi) {
-		methods[methods_count] = LOCATION_METHOD_WIFI;
-		methods_count++;
-	}
-#endif
-
-	if (!copy_cfg.no_data.neighbor_cell) {
-		methods[methods_count] = LOCATION_METHOD_CELLULAR;
-		methods_index_cellular = methods_count;
-		methods_count++;
-	}
-
-	location_config_defaults_set(&config, methods_count, methods);
+	/* Set default location configuration configured at compile time */
+	location_config_defaults_set(&config, 0, NULL);
 
 	config.timeout = copy_cfg.location_timeout * MSEC_PER_SEC;
 
-	if (methods_index_gnss != -1) {
-		config.methods[methods_index_gnss].gnss.timeout =
-			DATA_FETCH_TIMEOUT_GNSS_SEARCH * MSEC_PER_SEC;
-	}
+	/* If any method type has been disabled, update the method list by using those methods
+	 * that are enabled
+	 */
+	if (copy_cfg.no_data.neighbor_cell || copy_cfg.no_data.gnss || copy_cfg.no_data.wifi) {
+
+		for (int i = 0; i < config.methods_count; i++) {
+			/* Use methods that are not disabled at run-time in no_data configuration */
+			if ((config.methods[i].method == LOCATION_METHOD_GNSS &&
+				!copy_cfg.no_data.gnss) ||
+				(config.methods[i].method == LOCATION_METHOD_WIFI &&
+				!copy_cfg.no_data.wifi) ||
+				(config.methods[i].method == LOCATION_METHOD_CELLULAR &&
+				!copy_cfg.no_data.neighbor_cell)) {
+
+				memcpy(&methods_updated[methods_count],
+				       &config.methods[i],
+				       sizeof(struct location_method_config));
+				methods_count++;
+			}
+		}
```

Comment:
This is not particularly elegant way of combining compile-time and run-time configuration but I'm not sure there is a clean way of doing it. Especially if ATv2 wants to take advantage of other Kconfig default values such as `CONFIG_LOCATION_REQUEST_DEFAULT_GNSS_TIMEOUT`.

I need your feedback @simensrostad.

Reply:
That's what this code does together with `location_config_defaults_set()` call.
Alternative is to generate the method list in ATv2 side based on Location library Kconfig's (or create your own Kconfig's) which would mean we copy some code from Location library to ATv2. This is not particularly beautiful either.

And the reason is that we are kind of carrying the legacy of being able to configure the method `nod` list run-time while we should be really configuring the priority list. Or I would guess that if ATv2 was implemented now from scratch, that's what would be done when Location library would be used from the beginning.

Reply:
I see, how about making `location_config_defaults_set()` take arguments? Then there would be a single entry to prepare the `config` structure.

Reply:
Based on offline discussion:
We could add flags about "nod" list into `location_config_defaults_set()`. However, this is seen as ATv2 specific feature and as I commented in my previous response, ATv2 would probably be made differently today. Hence, we want to avoid this kind of application specific compromise which are not thought to be used more generally. ATv2 can handle the "nod" list as has been implemented although it's not particularly elegant.

Reply:
This piece of code is staying currently. I'm not sure any alternatives are any better given Location library won't have support for dynamic "nod" list.